### PR TITLE
Convert `modules/check-mediator` to ES6

### PR DESCRIPTION
**Pull request by gu-stav**
## What does this change?

Converts `modules/check-mediator` and specs to ES6.

## What is the value of this and can you measure success?

ES6!

## Does this affect other platforms - Amp, Apps, etc?

No.

## Screenshots

No.

## Tested in CODE?

No.

### CODE REVIEW

**Comment by gustavpursche** [comment]
@LATaylor-guardian I update the PR to reflect the changes you did in https://github.com/guardian/frontend/commit/28c2e59d7fac8b63f457900552d52eded5ff2f87 Please have a quick 👀 if everything looks good to you. 🙏

File: static/src/javascripts/projects/common/modules/check-mediator.js [hunk 1]
```diff
@@ -0,0 +1,111 @@
+// @flow
+
+import checks from './check-mediator-checks';
```
Comment (GHaberis) [inline static/src/javascripts/projects/common/modules/check-mediator.js]: Importing the checks is good, this means we can remove the _testRegisterCheck function and mock check-mediator-checks on the test.

File: static/src/javascripts/projects/common/modules/check-mediator-checks.js [hunk 2]
```diff
@@ -0,0 +1,123 @@
+// @flow
+
+/**
+ * referenced as passCondition for dependentChecks
+ * SOMECHECKSPASSED: At least one dependentCheck has returned true
+ * EVERYCHECKPASSED: Every dependentCheck has returned true
+**/
+const SOMECHECKSPASSED = Array.prototype.some;
```
Comment (regiskuckaertz) [inline static/src/javascripts/projects/common/modules/check-mediator-checks.js]: Geek comment: const { some: SOMECHECKSPASSED, every: EVERYCHECKPASSED } = Array.prototype;

**Comment by GHaberis** [comment]
@gustavpursche the es6 conversion looks good to me! I removed the testRegisterCheck function from the checkMediator module and mocked the 'checks' import in the checkMediator unit tests. I'm going to hold on making the updates to the check-dispatcher module in this PR and will make them once this has merged.

**Comment by gustavpursche** [comment]
@GHaberis Cool! Could you then approve the PR and then we get this 🚢 ed 💥. //cc @SiAdcock

**Review by GHaberis — APPROVED** [review]

**Pull request merged**